### ISSUE TITLE
Warlock talent Nightfall proc

### ISSUE THREAD

**Issue by whispersz** [SunstriderEmu/BugTracker#2353]
**Describe the bug**
If you are casting a Shadowbolt and during this cast you get a Nightfall proc, when the already casting shadowbolt is finished, it also consumes Nightfall proc.

**To Reproduce**
1. Spec into affliction and put 2/2 into Nightfall
2. cast Shadowbolts
3.  You will see that already casting shadowbolt will consume the nightfall proc that makes your next shadowbolt instant cast

Item: https://wowhead.com?item=
Quest: https://wowhead.com?quest=
Spell: https://classic.wowhead.com/spell=11661/shadow-bolt
https://classic.wowhead.com/spell=18095/nightfall
**Expected behavior**
When you cast a Shadowbolt and during this cast you get a Nighfall proc, when the cast time is finished and your shadobolt is sent to target, it should not consume your Nightfall proc.
Basically if you get a nightfall proc during shadowbolt cast, you can send 2 shadowbolts at once to the target(the one that you were already casting and the instant one from Nightfall proc, with global CD  of course) 

**Screenshots/videos**

**Additional context**

**Comment by blblblaz** [comment]
could not reproduce, https://streamable.com/0yh06, worked for me, need more details if possible

**Comment by whispersz** [comment]
I think it was a visual bug for me that both shadowbolts went at the same time and i could not see them. Indeed its working fine. 
Can be closed
